gradle/suppress.gradle

```diff
@@ -24,40 +23,53 @@ void suppressCves(reportName) {
                                     .collect { it.name.text() } as Set<String>
 
     if (!unSuppressedCves.isEmpty()) {
-        File suppressionsFile = project.file(dependencyCheck.getSuppressionFile())
-        def suppressionXml = new String(suppressionsFile.readBytes(), "UTF-8")
-        Element suppressionsDoc = DOMBuilder.parse(new StringReader(suppressionXml)).getDocumentElement()
-        addNewSuppressions(unSuppressedCves, suppressionsDoc)
-        suppressionsFile.write(XmlUtil.serialize(cleanXml(suppressionsDoc)))
+        // Suppression file must be specified as 'suppressionFile=...' in the build script.
+        File suppressionFile = project.file(dependencyCheck.getSuppressionFile())
+        String suppressionXml = new String(suppressionFile.readBytes(), "UTF-8")
+        Element suppressionDoc = DOMBuilder.parse(new StringReader(suppressionXml)).getDocumentElement()
+        addNewSuppression(unSuppressedCves, suppressionDoc)
+        suppressionFile.write(XmlUtil.serialize(cleanXml(suppressionDoc)))
     }
 }
 
-static void addNewSuppressions(unSuppressedCves, suppressions) {
-    unSuppressedCves.each {
-        String cveName = it
-        Document document = suppressions.getOwnerDocument()
-        Element suppressTag = document.createElement("suppress")
-        suppressTag.appendChild(document.createElement("notes")).appendChild(document.createTextNode("Ticket TBC"))
-        suppressTag.appendChild(document.createElement("cve")).appendChild(document.createTextNode(cveName))
-        suppressions.appendChild(suppressTag)
+static void addNewSuppression(unSuppressedCves, suppressionDoc) {
+    // Find 'Temporary Suppression' in note element and amend text content with new cve
+    XPathExpression xpathExp = XPathFactory.newInstance().newXPath().compile(
+            "//*[local-name()='notes']")
+    def notesNodes =  xpathExp.evaluate(suppressionDoc, XPathConstants.NODESET)
+    for (int i = 0; i < notesNodes.getLength(); i++) {
+        Node noteNode = notesNodes.item(i)
+        if (noteNode.getTextContent().contains('Temporary Suppression')) {
+            unSuppressedCves.each {
+                String whiteSpaceForIndent = "\n" + "\u0020".repeat(8) // white space for indentation
+                Document document = suppressionDoc.getOwnerDocument()
+                noteNode.appendChild(
+                        document.createTextNode(whiteSpaceForIndent + it + " refer [Ticket]"))
+                noteNode.getParentNode()
+                        .appendChild(document.createElement("cve")).appendChild(document.createTextNode(it))
+            }
+        }
     }
```

Comment:
1.) Might want to change the for loop to a while loop with an exit condition.  There's only going to be one notes section you want to update, so you can exit the loop after updating it.
2.) The value of the whiteSpaceForIndent variable never changes after it has been declared so move the declaration outside of the outer for loop.  That'll save it from be redeclared in each iteration of the inner loop.  Might also be an idea to make it a constant.
3.) The value of the document variable never changes after it has been declared so move the declaration outside of the outer for loop.
4.) Could create a variable called suppressNode as first line of if statement and set it to value of noteNode.getParentNode().  Use this in place of noteNode.getParentNode() on line 48.  This would avoid the need to call noteNode.getParentNode() on each iteration of the inner loop.

Reply:
1. done; uses find{} in replace of the for loop. Exists when true is returned
2. done; moved whiteSpaceForIndent outside the loop
3. since it's only needed once, it's left inside the if statement
4. done